lifetimes/models/beta_geo_model.py

```diff
@@ -0,0 +1,350 @@
+from __future__ import generator_stop
+from __future__ import annotations
+import warnings
+
+from typing import Union, Tuple
+
+import pandas as pd
+import numpy as np
+import numpy.typing as npt
+
+import pymc as pm
+import aesara.tensor as at
+
+from autograd.scipy.special import gammaln, beta, gamma
+from scipy.special import hyp2f1
+from scipy.special import expit
+
+from . import BaseModel
+from ..utils import _scale_time, _check_inputs
+from ..generate_data import beta_geometric_nbd_model
+
+
+class BetaGeoModel(BaseModel['BetaGeoModel']):
+    """
+    Also known as the BG/NBD model.
+    Based on [2]_, this model has the following assumptions:
+    1) Each individual, i, has a hidden lambda_i and p_i parameter
+    2) These come from a population wide Gamma and a Beta distribution
+       respectively.
+    3) Individuals purchases follow a Poisson process with rate lambda_i*t .
+    4) After each purchase, an individual has a p_i probability of dieing
+       (never buying again).
+
+    Attributes
+    ----------
+    params_: :obj: Series
+        The fitted parameters of the model
+    data: :obj: DataFrame
+        A DataFrame with the values given in the call to `fit`
+    attr_name : datatype
+        Add description here.
+        
+    References
+    ----------
+    .. [2] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005a),
+       "Counting Your Customers the Easy Way: An Alternative to the
+       Pareto/NBD Model," Marketing Science, 24 (2), 275-84.
+
+    """
+    
+    remove_hypers = ['BetaGeoModel::phi','BetaGeoModel::kappa']
+
+    def _model(self) -> pm.Model():
+
+        with pm.Model(name=f'{self.__class__.__name__}') as self.model:
+            # Priors for lambda parameters.
+            alpha_prior = pm.Weibull(name="alpha", alpha=1, beta=6)
+            r_prior = pm.Weibull(name="r", alpha=1, beta=1)
+
+            # Heirarchical pooling of hyperpriors for beta parameters.
+            phi_prior = pm.Uniform('phi', lower=0.0, upper=1.0)
+            kappa_prior = pm.Pareto('kappa',alpha=1,m=1.5)
```

Comment:
It would be nice to be able to allow the user to modify the priors, for example as in Bambi : https://bambinos.github.io/bambi/main/notebooks/radon_example.html

Reply:
Ups! I just read this would be part of the second iteration 🙈 !

Reply:
Yes indeed! The extent of modification that Bambi supports would take some time to implement, but making the prior distribution parameters arguments for `BetaGeo._model` is straightforward.